openff/interchange/interop/openmm/_topology.py

```diff
@@ -173,6 +186,26 @@ def to_openmm_topology(
                 order=bond_order,
             )
 
+    if has_virtual_sites and not collate:
+        # TODO: This structure is imperfect, no clue how to handle chains here
+        virtual_site_chain = openmm_topology.addChain(atom_chain_id)
+
+        for molecule_index, molecule in enumerate(topology.molecules):
+            virtual_sites_in_this_molecule: list[VirtualSiteKey] = molecule_virtual_site_map[molecule_index]
+
+            # make a new "residue" for each molecule which has virtual sites
+            if len(virtual_sites_in_this_molecule) > 0:
+                this_virtual_site_residue = openmm_topology.addResidue("VS", virtual_site_chain)
```

Comment:
If we're not picking up the resid from anywhere, I believe that doesn't need to be unique - so it might make sense to set the resid to of the virtual site residue to the same value of its parent residue?

Reply:
Sure - with a21f2ceeceab08e7587c4ef0ea0be8e4656005cd,

```python
In [1]: from openff.toolkit import *

In [2]: mol = Molecule.from_smiles("O")

In [3]: for atom in mol.atoms:
   ...:     atom.metadata["residue_name"] = "SOL"

In [4]: In [30]: x = (
   ...:     ...:     ForceField("opc.offxml")
   ...:     ...:     .create_interchange(
   ...:     ...:         Topology.from_molecules(
   ...:     ...:             [
   ...:     ...:                 mol,#Molecule.from_smiles("O"),
   ...:     ...:                 mol,#Molecule.from_smiles("O"),
   ...:                          mol,
   ...:     ...:             ],
   ...:     ...:         ),
   ...:     ...:     )
   ...:     ...:     .to_openmm_topology()
   ...:     ...: )
/Users/mattthompson/micromamba/envs/new-models/lib/python3.11/site-packages/smirnoff99frosst/smirnoff99frosst.py:11: UserWarning: Module openff was already imported from None, but /Users/mattthompson/software/openff-interchange is being added to sys.path
  from pkg_resources import resource_filename
/Users/mattthompson/micromamba/envs/new-models/lib/python3.11/site-packages/mdtraj/formats/__init__.py:13: DeprecationWarning: 'xdrlib' is deprecated and slated for removal in Python 3.13
  from mdtraj.formats.trr import TRRTrajectoryFile


In [5]: for residue in x.residues():
   ...:     print(residue.index, [*residue.atoms()])
0 [<Atom 0 (O1x) of chain 0 residue 0 (SOL)>, <Atom 1 (H1x) of chain 0 residue 0 (SOL)>, <Atom 2 (H2x) of chain 0 residue 0 (SOL)>]
1 [<Atom 3 (O1x) of chain 1 residue 1 (SOL)>, <Atom 4 (H1x) of chain 1 residue 1 (SOL)>, <Atom 5 (H2x) of chain 1 residue 1 (SOL)>]
2 [<Atom 6 (O1x) of chain 2 residue 2 (SOL)>, <Atom 7 (H1x) of chain 2 residue 2 (SOL)>, <Atom 8 (H2x) of chain 2 residue 2 (SOL)>]
3 [<Atom 9 (EP) of chain 3 residue 3 (SOL)>]
4 [<Atom 10 (EP) of chain 3 residue 4 (SOL)>]
5 [<Atom 11 (EP) of chain 3 residue 5 (SOL)>]

Reply:
Is that for my other comment 😅?

Reply:
Ah, yes, IDs and names are different nouns, after all ...